heat/core/linalg/basics.py

```diff
@@ -74,45 +82,88 @@ def cross(x1: DNDarray, x2: DNDarray, axis: int = -1) -> DNDarray:
     sanitation.sanitize_in(x1)
     sanitation.sanitize_in(x2)
 
-    # 2d -> 3d vector
-    if x1.shape[axis] == 2:
-        shape = tuple(1 if i == axis else j for i, j in enumerate(x1.shape))
-        x1 = manipulations.concatenate(
-            [x1, factories.zeros(shape, dtype=x1.dtype, device=x1.device)]
-        )
-
-    if x2.shape[axis] == 2:
-        shape = tuple(1 if i == axis else j for i, j in enumerate(x2.shape))
-        x2 = manipulations.concatenate(
-            [x2, factories.zeros(shape, dtype=x2.dtype, device=x2.device)]
-        )
-
-    if x1.gshape != x2.gshape:
-        raise ValueError(
-            "'x1' and 'x2' must have the same shape, {} != {}".format(x1.gshape, x2.gshape)
-        )
-    if x1.split != x2.split:
-        raise ValueError(
-            "'x1' and 'x2' must have the same split, {} != {}".format(x1.split, x2.split)
-        )
     if x1.device != x2.device:
         raise ValueError(
             "'x1' and 'x2' must have the same device type, {} != {}".format(x1.device, x2.device)
         )
     if x1.comm != x2.comm:  # pragma: no cover
         raise ValueError("'x1' and 'x2' must have the same comm, {} != {}".format(x1.comm, x2.comm))
 
-    if not isinstance(axis, int):
-        try:
-            axis = int(axis)
-        except Exception:
-            raise TypeError("'axis' must be an integer.")
+    x1_2d, x2_2d = False, False
+    x1_shape, x2_shape = list(x1.shape), list(x2.shape)
 
-    axis = stride_tricks.sanitize_axis(x1.shape, axis)
+    if not axis == -1 or torch.unique(torch.tensor([axisa, axisb, axisc, axis])).numel() == 1:
```

Comment:
If the default value of `axis` is set to `None`, the check will be shortened.

Reply:
`axis = None` is not compatible with the [Python array API](https://data-apis.org/array-api/latest/extensions/linear_algebra_functions.html?highlight=cross#linalg-cross-x1-x2-axis-1) though.

If we leave `axis=-1` as default, we're compatible with the array API and `cross` will still work as a numpy drop-in replacement.

Reply:
> If the default value of `axis` is set to `None`, the check will be shortened. Also, some redundant code can be avoided when the axes parameter are set beforehand to the value of it.
> 
> ```python
> if axis is not None:
>     axis = stride_tricks.sanitize_axis(x1.shape, axis)
>     axisa, axisb, axisc = (axis,) * 3
> ```

You're right about the redundant code though, I replaced it with your suggestion.